### PR TITLE
add requestID middleware

### PR DESCRIPTION
**Pull request by micbar**
## Description

We improve the logging of several http services. If possible and present, we now log the `X-Request-Id`.

## Motivation and Context

Better debuggability

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by ownclouders** [comment]
:boom: Acceptance test [<strong>Web-Tests-ocis-smoke-ocis-storage-1</strong>](https://drone.owncloud.com/owncloud/ocis/16657/57/1) failed. Further test are cancelled...

**Comment by mmattel** [comment]
Something to be added in the documentation?

**Comment by micbar** [comment]
> Something to be added in the documentation?

there is a ticket https://github.com/owncloud/docs-ocis/issues/266

File: services/ocdav/pkg/config/config.go [hunk 1]
```diff
@@ -71,14 +71,20 @@ type HTTPConfig struct {
...
+	Auth      Auth      `yaml:"auth"`
+	RequestID RequestID `yaml:"request_id"`
 }
 
 // Auth configures reva http auth middleware.
 type Auth struct {
 	CredentialsByUserAgent map[string]string `yaml:"credentials_by_user_agent"`
 }
 
+// RequestID configures reva http auth middleware.
```
Comment (butonic) [inline services/ocdav/pkg/config/config.go:83]:
```suggestion
// RequestID configures reva requestid middleware.
```

**Review by butonic — CHANGES_REQUESTED** [review]

File: services/ocdav/pkg/config/config.go [hunk 2]
```diff
@@ -71,14 +71,20 @@ type HTTPConfig struct {
 
 // Middleware configures reva middlewares.
 type Middleware struct {
-	Auth Auth `yaml:"auth"`
+	Auth      Auth      `yaml:"auth"`
+	RequestID RequestID `yaml:"request_id"`
```
Comment (butonic) [inline services/ocdav/pkg/config/config.go:75]: unused
```suggestion
	RequestID RequestID `yaml:"request_id"`
```

File: services/ocdav/pkg/config/config.go [hunk 3]
```diff
@@ -71,14 +71,20 @@ type HTTPConfig struct {
...
 // Auth configures reva http auth middleware.
 type Auth struct {
 	CredentialsByUserAgent map[string]string `yaml:"credentials_by_user_agent"`
 }
 
+// RequestID configures reva requestid middleware.
+type RequestID struct {
+	RequestID map[string]string `yaml:"requestid"`
+}
+
```
Comment (butonic) [inline services/ocdav/pkg/config/config.go:87]: unused
```suggestion
```

**Review by rhafer — APPROVED** [review]

**Comment by mmattel** [comment]
This needs a rebase to get it green because of a fix in CI

**Review by butonic — CHANGES_REQUESTED** [review]
there is no use in adding the config struct here as the ocdav micre service does not use the config options, yet

**Comment by butonic** [comment]
Actually, the whole Middleware config is unused. CredentialsByUserAgent  is not used by the micro ocdav service, yet: https://github.com/cs3org/reva/blob/1bc292816a55e7a83fd5967a5baa25a5899d5194/pkg/micro/ocdav/service.go#L142-L156

File: services/ocdav/pkg/config/config.go (hunk 3 above)
Comment (butonic) [inline services/ocdav/pkg/config/config.go:87]:
```suggestion
```

**Comment by butonic** [comment]
to clarify the requestid middleware is always enabled when using dhe microfied ocdav: https://github.com/cs3org/reva/blob/1bc292816a55e7a83fd5967a5baa25a5899d5194/pkg/micro/ocdav/service.go#L195-L199

**Review by butonic — APPROVED** [review]

**Review by rhafer — APPROVED** [review]

**Pull request merged**